module/validation/seal_validator.go

```diff
@@ -157,6 +158,18 @@ func (s *sealValidator) Validate(candidate *flow.Block) (*flow.Seal, error) {
 	// For each visited block collect: IncorporatedResults and block ID
 	forkCollector := func(header *flow.Header) error {
 		blockID := header.ID()
+		// TEMPORARY implementation (unpolished)
```

Comment:
What's temporary in this fix given that this PR is labeled final, and that it updates traversal logic?

Reply:
Ideally, I'd like to generalize the fork traversal logic (step 2 in https://github.com/onflow/flow-go/pull/1325). 

Though, you are right in that this PR _completely_ covers the edge case. So it is not a temporary fix (though could be polished more). Removed the "TEMPORARY" designation in https://github.com/onflow/flow-go/pull/1325/commits/6e2d6bbad115de7c6a30064a785180667d3da8ae.